src/cockpit/channels/filesystem.py

```diff
@@ -124,14 +124,14 @@ class FsReadChannel(GeneratorChannel):
 
     def do_yield_data(self, options: JsonObject) -> Generator[bytes, None, JsonObject]:
         path = get_str(options, 'path')
-        max_read_size = get_int(options, 'max_read_size', None)
+        max_read_size = get_int(options, 'max_read_size', 16 * 1024 * 1024)
 
         logger.debug('Opening file "%s" for reading', path)
 
         try:
             with open(path, 'rb') as filep:
                 buf = os.stat(filep.fileno())
-                if max_read_size is not None and buf.st_size > max_read_size:
```

Comment:
Hmm, did the C bridge support any way of saying "indefinite"? Like "-1"? For download in files we have no way of limiting. But if we change files now to give -1, it will fail with the old bridge. So I think for the time being the only thing files can do is to set it to MAXINT64 for the time being. Then we add support for -1 here, and at some point we can change files accordingly. WDYT?

Reply:
The C bridge never did, we never had big download support until cockpit-files and as files started with only supporting the Python bridge this just "worked".

Cockpit only had:

```
pkg/sosreport/sosreport.jsx:        max_read_size: 150 * 1024 * 1024,
```

Your proposal sounds good, it actually would unbreak our RHEL-8 testing was what I was thinking but it doesn't as we beiboot :-)

Reply:
Implemented your proposal and see the PR description for the cockpit-files PR